iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -416,12 +417,13 @@ private void commitTable(FileIO io, ExecutorService executor, OutputTable output
 
     Table table = null;
     String branchName = null;
-
+    Expression filterExpr = Expressions.alwaysTrue();
     for (JobContext jobContext : outputTable.jobContexts) {
       JobConf conf = jobContext.getJobConf();
       table = Optional.ofNullable(table).orElse(Catalogs.loadTable(conf, catalogProperties));
       branchName = conf.get(InputFormatConfig.OUTPUT_TABLE_SNAPSHOT_REF);
-
+      filterExpr = Expressions.and(filterExpr, (Expression) SessionStateUtil.getResource(conf,
```

Comment:
what would be a final filter in case of multiple jobContexts? multiple duplicate expressions joined with `and`?

Reply:
The following example creates 2 job contexts per update query:

Thread 1:  UPDATE customers SET first_name='Changed' WHERE  last_name='Taylor'

Thread 2:  UPDATE customers SET first_name='Changed' WHERE  last_name='Donnel'

In the previous commit it would have multiple duplicate filters with AND. But i this commit, i have added a if condition to update the the filter only if they are different. This should help minimise the duplicate filters as seen in image below.

<img width="1624" alt="image" src="https://github.com/apache/hive/assets/26441677/fccb5ecb-8fe1-48d1-9e14-ecbc222961af">

Reply:
could you please check the filter construction for merge_iceberg_partitioned_orc.q 
````
merge into target_ice as t using source src ON t.a = src.a
when matched and t.a > 100 THEN DELETE
when matched then update set b = 'Merged', c = t.c + 10
when not matched then insert values (src.a, src.b, src.c);
````